### DISCUSSION TITLE
Typechain with Wagmi useContractRead/Write

### DISCUSSION THREAD

**Discussion by pmespresso** [wevm/wagmi#853] [Q&A] [answered]
I'm used to feeding Typechain interfaces into ethersjs and having all the function signature and return types ready for me in my Typescript React projects.

First time using wagmi-js for my current project, I was able to do the same with `useContract` however it seems that hook is discouraged. I want to do the same with `useContractWrite` and `useContractRead` but it doesn't seem these functions use generic types.

Is this just undocumented, or does it need to be implemented/ can it be impleneted? thank you

**Comment by tmm** [comment, marked as answer]
Coming to wagmi soon in https://github.com/wagmi-dev/wagmi/pull/941! Powered by the [ABIType](https://github.com/wagmi-dev/abitype).
